31Aug_HappyNumber.py

```diff
@@ -0,0 +1,46 @@
+#
+# https://leetcode.com/problems/happy-number/description/
+#
+#####################################################################################################################################################
+#                                                            PROBLEM DESCRIPTION                                                                    #
+#####################################################################################################################################################
+#
+# Write an algorithm to determine if a number n is happy.
+# A happy number is a number defined by the following process:
+#
+# Starting with any positive integer, replace the number by the sum of the squares of its digits.
+# Repeat the process until the number equals 1 (where it will stay), or it loops endlessly in a cycle which does not include 1.
+# Those numbers for which this process ends in 1 are happy.
+# Return true if n is a happy number, and false if not.
+#
+# Example 1:
+# Input: n = 19
+# Output: true
+# Explanation:
+# 12 + 92 = 82
+# 82 + 22 = 68
+# 62 + 82 = 100
+# 12 + 02 + 02 = 1
+#
+# Example 2:
+# Input: n = 2
+# Output: false
+#
+class Solution:
+    def isHappy(self, n: int) -> bool:
+        # n = 19
+        seen = []
```

Comment:
This should be a `set` which can do membership lookup in constant time, lists need to look in every item

Reply:
Code is changed to handle as set() as below:

```
class Solution:
    def isHappy(self, n: int) -> bool:
        # n = 19
        seen = set()
        while n not in seen:
            seen.add(n)
```

And there is a significant improvement in time in ms as in the attachment.
<img width="1469" alt="Screenshot 2023-09-04 at 12 31 45 PM" src="https://github.com/RevathiKannanGithub/leetcode/assets/37206283/49b5b2e8-dfea-4e39-a215-8f6d6c2d539a">